### PR TITLE
Fix 10947

### PR DESCRIPTION
**Pull request by radare**

### CODE REVIEW

File: binr/radare2/radare2.c [hunk 1]
```diff
@@ -1046,6 +1046,9 @@ int main(int argc, char **argv, char **envp) {
 				while (optind < argc) {
 					pfile = argv[optind++];
 					fh = r_core_file_open (&r, pfile, perms, mapaddr);
+					if (fh && (perms & R_IO_WRITE)) {
```
Comment (ret2libc) [inline binr/radare2/radare2.c]: why this change? Can't I write to a file with `va` enabled?

**Comment by radare** [comment]
try it yourself. already discussed with condret and tested it, seems the simpler solution

its in a pr for testing/discussion, also, this may only happen when opening a new file

and no, you cant write in va if there are no maps. and if the fuile is mapped you cant write beyond the end of it without resizing it. so this is what makes the behaviour expected

**Comment by ret2libc** [comment]
I think that if we don't allow writing with va enabled, then I see few reasons to use radare2 to write to binaries instead of using just an hex editor.

What I personally always found very useful in radare2 is that I can modify the program as I see it and not by opening it without the mappings.

**Comment by ret2libc** [comment]
Ok ignore my last comment.

File: binr/radare2/radare2.c (hunk 1 above)
Comment (ret2libc) [reply binr/radare2/radare2.c]: However, #10947 says that `io.va=false` should be done only when the file does not exist. `if (fh && (perms & R_IO_WRITE))` seems to be executed even when the file does exist, so I'm not sure about this.

**Comment by radare** [comment]
moved to 2.9 to have proper time for testing and discussing. and maybe we should have a flag in r2 to enable or disable va

File: binr/radare2/radare2.c (hunk 1 above)
Comment (kazarmy) [reply binr/radare2/radare2.c]: Probably just missing a bang (`!`).

Comment (ret2libc) [reply binr/radare2/radare2.c]: Nope, I checked and even non-existing files have a `fh` object.

**Comment by Maijin** [comment]
image

File: binr/radare2/radare2.c (hunk 1 above)
Comment (kazarmy) [reply binr/radare2/radare2.c]: r2 helpfully creates a new file if it doesn't exist so I guess that's where a `new_file` flag should be set or something similar.

Comment (kazarmy) [reply binr/radare2/radare2.c]: ... and it creates it before the prompt appears rather than on exit

**Comment by radare** [comment]
it's green now
